includes/license.php

```diff
@@ -0,0 +1,181 @@
+        $key = get_option( 'sws_license_key' );
+    }
+
+    if ( ! empty( $key ) ) {
+        $url = add_query_arg( array( 'license' => $key, 'domain' => site_url() ), SS_LICENSE_SERVER );
+
+        $timeout = apply_filters( 'sws_license_check_key_timeout', 5 );
+        $r = wp_remote_get( $url, array( 'timeout' => $timeout ) );
+
+        if ( is_wp_error( $r ) ) {
+           /// Handle errors here.
+        } elseif ( ! empty( $r ) && wp_remote_retrieve_response_code( $r ) == 200 ) {
+
+            $r = json_decode( $r['body'] );
+
+            if ( $r->active == 1 ) {
+
+				if ( ! empty( $r->enddate ) ) {
+					$enddate = strtotime( $r->enddate, current_time( 'timestamp' ) );
+                } else {
+					$enddate = strtotime( '+1 Year', current_time( 'timestamp' ) );
+                }
+
+				delete_option('sws_license_check');
+				add_option('sws_license_check', array( 'license' => $r->license, 'enddate' => $enddate), NULL, 'no');	
+				return true;
+			} elseif ( ! empty( $r->error ) ) {
+				//invalid key
+                /// Handle errors here.				
+				delete_option('sws_license_check');
+				add_option('sws_license_check', array('license'=>false, 'enddate'=>0), NULL, 'no');
+                
+			}
+        }
+    }
+}
+add_action( 'sws_license_check_key', 'sws_license_check_key' );
+
+/**
+ * Check if license prompt is paused.
+ */
+function sws_license_pause() {
+    if ( ! empty($_REQUEST['sws_nag_paused'] ) && current_user_can( 'manage_options' ) ) {
+		$sws_nag_paused = current_time( 'timestamp' ) + ( 3600*24*7 );
+		update_option('sws_nag_paused', $sws_nag_paused, 'no');
+		
+		return;
+    }
+}
+add_action( 'admin_init', 'sws_license_pause' );
+
+/**
+ * Add license nag to plugin admin page headers.
+ */
+function sws_license_nag() {
+    static $sws_nagged;
+
+    // Did we already nag the user?
+    if ( ! empty( $sws_nagged ) ) {
+        return;
+    }
+
+    $sws_nagged = true;
+
+    // Blocked by constant
+    if ( defined( 'SWS_LICENSE_NAG' ) && ! SWS_LICENSE_NAG ) {
+        return;
+    }
+
+    if ( empty( $_REQUEST['post_type'] ) ) {
+        return;
+    }
+
+    if ( $_REQUEST['post_type'] !== 'sitewide_sale') {
+        return;        
+    }
+
+    // Don't load nag on license page.
+    if ( ! empty( $_REQUEST['page'] ) && $_REQUEST['page'] === 'sitewide_sales_license' ) {
+        return;
+    }
+
+    // Bail if they have a valid license already.
+    if ( sws_license_is_valid() ) {
+        return;
+    }
+
+    $sws_nag_paused = get_option( 'sws_nag_paused', 0 );
+
+    if ( current_time( 'timestamp' ) < $sws_nag_paused && $sws_nag_paused < current_time( 'timestamp' ) * 3600*24*35 ) {
+        return;
+    }
+
+    // Get the key to use later.
+    $key = get_option( 'sws_license_key' );
+
+    //okay, show nag
+    ?>
+    <div class="<?php if(!empty($key)) { ?>sitewide_sales_message sitewide_sales_error<?php } else { ?>sitewide_sales_message sitewide_sales_error<?php } ?> fade">		
```

Comment:
Both cases of `if` statement see to do the same thing

Reply:
This has been updated.